OutOfSchool/OutOfSchool.Common/Validators/CustomAgeAttribute.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace OutOfSchool.Common.Validators;
+
+/// <summary>
+/// Validation attribute for dates. It allows only dates that are at least <see cref="MinAge"/> and at most <see cref="MaxAge"/>.
+/// This attribute can be used on <see cref="DateTime"/> or <see cref="DateOnly"/> types.
+/// </summary>
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = false)]
+public class CustomAgeAttribute : DataTypeAttribute
+{
+    /// <summary>
+    /// The approximate number of days in a year including leap years.
+    /// </summary>
+    public const double ApproximateDaysInYear = 365.2425;
+
+    public CustomAgeAttribute()
+        : base(DataType.Date)
+    {
+    }
+
+    /// <summary>
+    /// Gets the minimum age of the persom in years.
+    /// </summary>
+    public int MinAge { get; init; } = 0;
+
+    /// <summary>
+    /// Gets the maximum age of the persom in years.
+    /// </summary>
+    public int MaxAge { get; init; } = int.MaxValue;
+
+    /// <summary>
+    /// Gets a value indicating whether to use UTC now as beginning of time or not. Default value is <c>true</c>.
+    /// </summary>
+    public bool UseUTC { get; init; } = true;
+
+    /// <summary>
+    /// Gets the date time kind based on <see cref="UseUTC"/> value.
+    /// </summary>
+    private DateTimeKind DateTimeKind => UseUTC ? DateTimeKind.Utc : DateTimeKind.Local;
```

Comment:
> (Design) I'm not insisting on redesign, you could leave it as is.
> 
> Personally me would avoid such things
> 
> * private property - could be field or method;
> * getter-only property (that could never have a setter) - could be method;
> * simple private class member used in only one place - could be inlined.

Got it!
But one question: what is wrong with getter-only property? Same with compute-only property `int Prop => a + b` ?

Reply:
> But one question: what is wrong with getter-only property?

TLDR; It depends ) 

Usually they work in a same way as method, but promise more, provide wider API / potential for changes and thus need to be maintained (both in implementation and usage sites) more carefully. Also semantic is different. And usage.

---

Talking about "wider API" I mean that (in pseudocode):
`public int SomeNumber => SomeCodeToGetNumber();` 
could be changed to
`public int SomeNumber { get => SomeCodeToGetNumber(); private set; }`, 
that could be changed to
`public int SomeNumber { get => SomeCodeToGetNumber(); private set => DoNothingSpecial(value); }`, 
and later in next version of code or descendants it could be changed to (_setter_ is changed)
`public int SomeNumber { get => SomeCodeToGetNumber(); private set => DoSomething(value); }`, 
or even (_public_ added)
`public int SomeNumber { get => SomeCodeToGetNumber(); public set => DoSomething(value); }`.

Consumer could left such changes unnoticed (you even don't need to recompile consumer code), but logic can by drastically and (more important) unexpectedly changed. That's why, though you usually don't need/use these extension capabilities, you still should take them into account/support them for properties. 

Methods - have more limited "API" and potential for extensions and thus easier for support/change.

---

Also there is semantical difference, as property is something that could be set (independently), it could be hidden but still "settable", while method can't be "set".

I.e. let's check imaginary `Shape` class, calculated property `public int Area => Width * Height`. 

Could you set it? In generic case - no, you should do this via related properties. Ofc, there could be cases with some logic for "set" (like for circles/squares/rectangles with fixed aspect ratio) but to make that "set" isolated and not to break class invariant it is better to either have immutable class and construct a new `Shape` with new `Width` and `Height` or (for mutable class case) have specific `SetDimensions(int width, int height)` method.

---

Usage. As a "rule of thumb" class properties are main target for de/serialization, mapping, input validation, etc. Sometimes you can't control such activities and need something as property even if it breaks beforementioned concerns.

---

So, in general it depends. But not for this specific `DateTimeKind` case.

Reply:
> > But one question: what is wrong with getter-only property?
> 
> TLDR; It depends )
> 
> Usually they work in a same way as method, but promise more, provide wider API / potential for changes and thus need to be maintained (both in implementation and usage sites) more carefully. Also semantic is different. And usage.
> 
> Talking about "wider API" I mean that (in pseudocode): `public int SomeNumber => SomeCodeToGetNumber();` could be changed to `public int SomeNumber { get => SomeCodeToGetNumber(); private set; }`, that could be changed to `public int SomeNumber { get => SomeCodeToGetNumber(); private set => DoNothingSpecial(value); }`, and later in next version of code or descendants it could be changed to (_setter_ is changed) `public int SomeNumber { get => SomeCodeToGetNumber(); private set => DoSomething(value); }`, or even (_public_ added) `public int SomeNumber { get => SomeCodeToGetNumber(); public set => DoSomething(value); }`.
> 
> Consumer could left such changes unnoticed (you even don't need to recompile consumer code), but logic can by drastically and (more important) unexpectedly changed. That's why, though you usually don't need/use these extension capabilities, you still should take them into account/support them for properties.
> 
> Methods - have more limited "API" and potential for extensions and thus easier for support/change.
> 
> Also there is semantical difference, as property is something that could be set (independently), it could be hidden but still "settable", while method can't be "set".
> 
> I.e. let's check imaginary `Shape` class, calculated property `public int Area => Width * Height`.
> 
> Could you set it? In generic case - no, you should do this via related properties. Ofc, there could be cases with some logic for "set" (like for circles/squares/rectangles with fixed aspect ratio) but to make that "set" isolated and not to break class invariant it is better to either have immutable class and construct a new `Shape` with new `Width` and `Height` or (for mutable class case) have specific `SetDimensions(int width, int height)` method.
> 
> Usage. As a "rule of thumb" class properties are main target for de/serialization, mapping, input validation, etc. Sometimes you can't control such activities and need something as property even if it breaks beforementioned concerns.
> 
> So, in general it depends. But not for this specific `DateTimeKind` case.

Wow thanks! I never thought about such problems with properties, as for my PR I already removed `DateTimeKind`.